OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -271,7 +271,24 @@ public virtual IEnumerable<ActorInfo> BuildableItems()
 			if (developerMode.AllTech)
 				return Producible.Keys;
 
-			return buildableProducibles;
+			return buildableProducibles.Where(a => CheckBuildLimit(a));
+		}
+
+		public bool CheckBuildLimit(ActorInfo actor)
```

Comment:
`BuildableItems()` is called from several places in the sidebar code, so this would run a few hundred times per second. We really don't want to be doing significant logic or trait queries here!

Reply:
Ok,  marked for tomorrow.

Reply:
Do you have any idea on where to put this check? I think I all my ideas now is to put the build limit check back to techtree which is back to the old version (or I have to say the fix fails)

Reply:
Like I said in #18100, I think it is a mistake to try and touch anything related to the tech tree or this bug until we are ready to commit to the full proper rewrite that the code needs.

The icon issue is a minor cosmetic bug that has no impact on gameplay, and I do not think it is worth wasting the many hours of review and testing time that I expect it will take to fix the regressions that any changes would surely introduce.

Reply:
ok